File: test/fixtures/helloworld-python/helloworld/greet.py
"""Simple greetings. Every path here is covered by tests/test_greet.py."""


def hello(name: str = "World") -> str:
    """Return the classic greeting for one name."""
    if not name:
        name = "World"
    return f"Hello, {name}!"


def hello_many(names: list[str]) -> str:
    """Greet several people in one line."""
    if not names:
        return hello()
    if len(names) == 1:
        return hello(names[0])
    return f"Hello, {', '.join(names[:-1])} and {names[-1]}!"


def shout(text: str) -> str:
    """Make a greeting louder."""
    return text.upper().replace("!", "!!!")

File: test/fixtures/helloworld-react-playwright-ct/src/greet.ts
/** Simple greetings. Every path here is covered by greet.test.ts. */

export function hello(name = 'World'): string {
  if (!name) {
    name = 'World';
  }
  return `Hello, ${name}!`;
}

export function helloMany(names: string[]): string {
  if (names.length === 0) {
    return hello();
  }
  if (names.length === 1) {
    return hello(names[0]);
  }
  return `Hello, ${names.slice(0, -1).join(', ')} and ${names[names.length - 1]}!`;
}

export function shout(text: string): string {
  return text.toUpperCase().replace(/!/g, '!!!');
}
